docs/source/typevars.py

```diff
@@ -0,0 +1,101 @@
+"""Transform references to typevars to avoid missing reference errors.
+
+See https://github.com/sphinx-doc/sphinx/issues/7722 also.
+"""
+from __future__ import annotations
+
+import re
+
+from pathlib import Path
+
+from sphinx.addnodes import pending_xref, Element
+from sphinx.application import Sphinx
+from sphinx.environment import BuildEnvironment
+from sphinx.errors import NoUri
+
+
+def identify_typevars() -> None:
+    """Record all typevars in trio."""
+    trio_folder = Path(__file__, "..", "..", "..", "trio").resolve()
```

Comment:
could maybe query `git rev-parse --show-toplevel` for the root directory instead of hardcoding this monstrosity, like in
https://github.com/Zac-HD/shed/blob/73ea9159c5cec2c462ce0779457231dc69a79466/src/shed/_cli.py#L26-L34

Reply:
Hmm actually just `trio.__file__` too, already got that imported. Ideally we’d find where Sphinx stores the path, use that…